### PR TITLE
[BISERVER-14523] New Mondrian Catalog Cache Supporting Incremental ca…

### PR DESCRIPTION
**Pull request by tkafalas**
…ching

### CODE REVIEW

**Comment by buildguy** [comment]
### :warning: Build finished in 5m 48s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.platform.dataaccess.datasource.api.AnalysisServiceTest.testImportingSchemaRemovesExistingAnnotationsByDefault</b> (click to expand)</summary><p>

```

Wanted but not invoked:
iAclAwareMondrianCatalogService.removeCatalog(
    "sales",
    <any>
);
-> at org.pentaho.platform.dataaccess.datasource.api.AnalysisServiceTest.testImportingSchemaRemovesExistingAnnotationsByDefault(AnalysisServiceTest.java:134)

However, there were other interactions with this mock:
iAclAwareMondrianCatalogService.getCatalog(
    "sales",
    null
);
-> at org.pentaho.platform.dataaccess.datasource.api.AnalysisService.processMondrianImport(AnalysisService.java:251)


```

</p></details>

  
Tests run: **344**, Failures: **1**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/data-access-plugin/job/PR-1099/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 6m 38s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **344**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/data-access-plugin/job/PR-1099/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 6m 26s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **344**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/data-access-plugin/job/PR-1099/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: core/src/main/java/org/pentaho/platform/dataaccess/datasource/api/AnalysisService.java [hunk 1]
```diff
@@ -248,12 +248,9 @@ protected void processMondrianImport( InputStream dataInputStream, String catalo
     if ( isOverwriteAnnotations( parameters, overWriteInRepository ) ) {
       IMondrianCatalogService catalogService =
           PentahoSystem.get( IMondrianCatalogService.class, PentahoSessionHolder.getSession() );
-      List<MondrianCatalog> catalogs = catalogService.listCatalogs( PentahoSessionHolder.getSession(), false );
-      for ( MondrianCatalog catalog : catalogs ) {
```
Comment (lucboudreau) [inline core/src/main/java/org/pentaho/platform/dataaccess/datasource/api/AnalysisService.java:252]: 🤦‍♂️

**Review by lucboudreau — APPROVED** [review]

**Comment by ddiroma** [comment]
Approved

**Pull request merged**